web/src/components/tasks/business-formation/BusinessNameSection.tsx

```diff
@@ -0,0 +1,173 @@
+import { Content } from "@/components/Content";
+import { Alert } from "@/components/njwds-extended/Alert";
+import { Button } from "@/components/njwds-extended/Button";
+import { Icon } from "@/components/njwds/Icon";
+import { useBusinessNameSearch } from "@/lib/data-hooks/useBusinessNameSearch";
+import { useUserData } from "@/lib/data-hooks/useUserData";
+import { MediaQueries } from "@/lib/PageSizes";
+import { SearchBusinessNameError } from "@/lib/types/types";
+import analytics from "@/lib/utils/analytics";
+import { templateEval } from "@/lib/utils/helpers";
+import Defaults from "@businessnjgovnavigator/content/display-defaults/defaults.json";
+import { FormControl, TextField, useMediaQuery } from "@mui/material";
+import React, { ReactElement, useContext } from "react";
+import { FormationContext } from "../BusinessFormation";
+
+const SearchBusinessNameErrorLookup: Record<SearchBusinessNameError, string> = {
+  BAD_INPUT: Defaults.searchBusinessNameTask.errorTextBadInput,
+  SEARCH_FAILED: Defaults.searchBusinessNameTask.errorTextSearchFailed,
+};
+
+export const BusinessNameSection = (): ReactElement => {
+  const { state, setTab, setFormationFormData } = useContext(FormationContext);
+  const { userData, update } = useUserData();
+  const isMobile = useMediaQuery(MediaQueries.isMobile);
+  const {
+    currentName,
+    submittedName,
+    isNameFieldEmpty,
+    isLoading,
+    error,
+    nameAvailability,
+    updateButtonClicked,
+    updateCurrentName,
+    onBlurNameField,
+    searchBusinessName,
+    updateNameOnProfile,
+  } = useBusinessNameSearch(true);
+
+  const submitNameAndContinue = async () => {
+    if (!userData) return;
+    const newFormationFormData = {
+      ...state.formationFormData,
+      businessName: submittedName,
+    };
+    setFormationFormData(newFormationFormData);
```

Comment:
This felt a little weird to update both the local (under FormationContext) state and the user data with the same value at the same time. But I thought it's better than updating the local FormationContext state every time the name input changes, since that would trigger a re-render of the entire `<BusinessFormation />` each time. Instead, I use local non-Context state in this component to handle input changes, and only update the Context one upon submit.